src/TwigComponent/src/ComponentAttributes.php

```diff
@@ -0,0 +1,100 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\UX\TwigComponent;
+
+/**
+ * @author Kevin Bond <kevinbond@gmail.com>
+ *
+ * @experimental
+ */
+final class ComponentAttributes
+{
+    /** @var array<string, string> */
+    public array $attributes;
+
+    /**
+     * @param array<string, string> $attributes
+     */
+    public function __construct(array $attributes)
+    {
+        $this->attributes = $attributes;
+    }
+
+    public function __toString(): string
```

Comment:
were you still thinking about adding validation for non-scalar values? (Not sure if it should live here, just using this as a spot for the comment)

Reply:
There is validation but in the `HasAttributesTrait`. Not ideal but improves the error message if misused. I didn't think we should validate twice but what do you think?

Reply:
It's good there - I forgot about it